cetmix_tower_server/models/cx_tower_file.py

```diff
@@ -132,6 +132,43 @@ class CxTowerFile(models.Model):
         compute="_compute_variable_ids",
         store=True,
     )
+    secret_ids = fields.Many2many(
+        comodel_name="cx.tower.key",
+        relation="cx_tower_secret_file_rel",
+        column1="file_id",
+        column2="secret_id",
+        string="Secrets",
+        readonly=True,
+        store=True,
+        compute="_compute_secret_ids",
+        groups="cetmix_tower_server.group_manager,cetmix_tower_server.group_root",
+    )
+
+    @api.depends("code")
+    def _compute_secret_ids(self):
+        """
+        Compute the secret IDs based on the references found in the command's code.
+        """
+        # Fetch all secrets and create a dictionary for quick lookup
+        secrets = self.env["cx.tower.key"].search([])
+        # we have reference_code field but it is False when secret type is [SSH key],
+        # so we use reference field
+        reference_to_secret = {
+            f"#!cxtower.secret.{secret.reference}!#": secret.id for secret in secrets
+        }
+
+        for command in self:
+            if command.code:
+                # Use a set for faster membership testing
+                code_references = set(command.code.split())
+                secret_ids = [
+                    reference_to_secret[ref]
+                    for ref in code_references
+                    if ref in reference_to_secret
+                ]
+                command.secret_ids = [(6, 0, secret_ids)]
+            else:
+                command.secret_ids = False
```

Comment:
Duplicating the same method across two models increases maintenance overhead and reduces code clarity. Instead, create a single shared method in a mixin or in "cx.tower.key" model and call it from both models. This ensures consistency and makes future updates easier.

Reply:
Thanks for review
I'll fix it